### PR TITLE
If 524 performance historical view date issue

### PR DESCRIPTION
**Pull request by g1glh0rn**

### CODE REVIEW

**Comment by mattmikulina** [comment]
Task linked: [IF-524 Historical View - Custom Date: Wrong data is shown](https://app.clickup.com/t/2177764/IF-524)

**Review by jsylupp — APPROVED** [review]
sieht gut aus

File: fusionfrontend/src/app/core/helpers/asset-chart-helper.ts [hunk 1]
```diff
@@ -26,4 +27,27 @@ export class AssetChartHelper {
...
+    let stepSize = 0;
+    if (axisUnit === 'day') {
+      stepSize = 12;
+    } else if (axisUnit === 'hour') {
+      stepSize = 3;
+    }
+    if (data.length === 0) {
+      return Date.now().valueOf();
+    }
+    const dates = data.map(datum => datum.t as number);
```
Comment (jsylupp) [inline fusionfrontend/src/app/core/helpers/asset-chart-helper.ts]: german word datum?

File: fusionfrontend/src/app/core/helpers/asset-chart-helper.ts [hunk 2]
```diff
@@ -26,4 +27,27 @@ export class AssetChartHelper {
...
+    const minDate = new Date(Math.min(...dates));
+    return minDate.setHours(minDate.getHours() - stepSize);
+  }
+
+  public static getMaxDateForLineChart(data: ChartPoint[]): number {
+    if (data.length === 0) {
+      return Date.now().valueOf();
+    }
+    const dates = data.map(datum => datum.t as number);
```
Comment (jsylupp) [inline fusionfrontend/src/app/core/helpers/asset-chart-helper.ts]: german word datum?

File: fusionfrontend/src/app/factory/components/content/asset-details/asset-charts/asset-charts.component.ts [hunk 3]
```diff
@@ -83,51 +85,152 @@ export class AssetChartsComponent implements OnInit, OnChanges, OnDestroy {
...
-    const minMaxYAxis =  AssetChartHelper.getYMinMaxByAbsoluteThreshold(this.fieldDetails);
+    const minMaxYAxis = AssetChartHelper.getYMinMaxByAbsoluteThreshold(this.fieldDetails);
+
+    let xAxisStepSize = 6;
+    let xAxisUnit: TimeUnit = 'hour';
+    let minDate = AssetChartHelper.getMinDateForLineChart(this.lineChartData[0].data as ChartPoint[], null);
+    const maxDate = AssetChartHelper.getMaxDateForLineChart(this.lineChartData[0].data as ChartPoint[]);
+
+    const dayInMillisecs = 24 * 60 * 60 * 1000;
+    if (maxDate - minDate > 5 * dayInMillisecs) {
```
Comment (jsylupp) [inline fusionfrontend/src/app/factory/components/content/asset-details/asset-charts/asset-charts.component.ts:206]: I guess 5 is assessed by viewing data right?

**Review by jsylupp — APPROVED** [review]
Looks good in my opinion. You can decide what to do with the comments

File: fusionfrontend/src/app/factory/components/content/asset-details/asset-charts/asset-charts.component.ts (hunk 3 above)
Comment (g1glh0rn) [reply fusionfrontend/src/app/factory/components/content/asset-details/asset-charts/asset-charts.component.ts:206]: yes, just a tradeoff. When having data for more than 5 days, it seems more reasonable to have axis ticks in a 24h interval at minimum, while for less than 5 days 6h at minimum is more suitable.

**Pull request merged**